jax/_src/lax/control_flow.py

```diff
@@ -1954,7 +1954,23 @@ def _check_tree_and_avals(what, tree1, avals1, tree2, avals2):
         f"{tree_unflatten(tree2, avals2)}.")
 
 
-def _check_tree(func_name, expected_name, actual_tree, expected_tree):
+def _check_tree(func_name, expected_name, actual_tree, expected_tree, has_aux=False):
+  if has_aux:
+    actual_tree_children = actual_tree.children()
+    num_children = len(actual_tree_children)
+
+    if num_children == 2:
+      # select first child as result tree if b is a singleton
+      actual_tree = tree_structure(actual_tree_children[0])
+    elif num_children > 2:
+      # prune away only the last child (aux)
+      actual_tree = tree_structure(actual_tree_children[:-1])
```

Comment:
When could this happen? I thought the return value should always be a tuple of two elements if `has_aux=True`.

Reply:
You are absolutely correct. I was not aware that the tuple is a pytree itself, and thought you had to operate on the tree structure of `b` there. I just cleaned this up in 0308527.